src/plugins/nvme/nvme.h

```diff
@@ -0,0 +1,327 @@
+ *                      attributes and a format (excluding secure erase) of any namespace results in
+ *                      a format of all namespaces in an NVM subsystem, %FALSE if the controller supports
+ *                      format on a per namespace basis.
+ * @feat_ns_mgmt: %TRUE if the controller supports the Namespace Management and Attachment capability.
+ * @feat_selftest: %TRUE if the controller supports the Device Self-test command.
+ * @feat_one_selftest: %TRUE if the NVM subsystem supports only one device self-test operation in progress at a time.
+ * @selftest_ext_time: Extended Device Self-test Time, if @feat_selftest is supported then this field
+ *                     indicates the nominal amount of time in one minute units that the controller takes
+ *                     to complete an extended device self-test operation when in power state 0.
+ * @hmb_pref_size: Host Memory Buffer Preferred Size indicates the preferred size that the host
+ *                 is requested to allocate for the Host Memory Buffer feature in bytes.
+ * @hmb_min_size: Host Memory Buffer Minimum Size indicates the minimum size that the host
+ *                is requested to allocate for the Host Memory Buffer feature in bytes.
+ * @size_total: Total NVM Capacity in the NVM subsystem in bytes.
+ * @size_unalloc: Unallocated NVM Capacity in the NVM subsystem in bytes.
+ * @feat_sanitize_crypto: %TRUE if the controller supports the Crypto Erase sanitize operation.
+ * @feat_sanitize_block: %TRUE if the controller supports the Block Erase sanitize operation.
+ * @feat_sanitize_overw: %TRUE if the controller supports the Overwrite sanitize operation.
+ * @feat_secure_erase_all: %TRUE in case any secure erase performed as part of a format operation
+ *                         results in a secure erase of all namespaces in the NVM subsystem,
+ *                         %FALSE in case any secure erase performed as part of a format results
+ *                         in a secure erase of the particular namespace specified.
+ * @feat_secure_erase_crypto: %TRUE if cryptographic erase is supported.
+ * @num_namespaces: Maximum Number of Allowed Namespaces supported by the NVM subsystem.
+ * @subsysnqn: NVM Subsystem NVMe Qualified Name, UTF-8 null terminated string.
+ */
+typedef struct BDNVMEControllerInfo {
+    gboolean feat_multiport;
+    gboolean feat_multictrl;
+    gboolean feat_sriov;
+    gboolean feat_ana_supported;
+    guint16 ctrl_id;
+    gchar *fguid;
+    gboolean feat_format;
+    gboolean feat_format_all_ns;
+    gboolean feat_ns_mgmt;
+    gboolean feat_selftest;
+    gboolean feat_one_selftest;
+    gint selftest_ext_time;
+    guint64 hmb_pref_size;
+    guint64 hmb_min_size;
+    guint64 size_total;
+    guint64 size_unalloc;
+    gboolean feat_sanitize_crypto;
+    gboolean feat_sanitize_block;
+    gboolean feat_sanitize_overw;
+    gboolean feat_secure_erase_all;
+    gboolean feat_secure_erase_crypto;
```

Comment:
It might be better to make all these "features" flags instead of booleans. Thoughts?

Reply:
Changed to enums.